sources/common/test/unitTest/ISignal.cpp

```diff
@@ -42,11 +42,12 @@ static inline std::string getISignalPrint(const std::string &value)
 static inline SIGNAL::ISignal getISignal(uint64_t value,
                                          size_t size = DummySignal::DUMMY_SIZE)
 {
-    return {DummySignal::DUMMY_NAME,
-            size,
+    static std::vector<std::unique_ptr<const SIGNAL::SignalDescriptor>> vDesc = {};
+    vDesc.push_back(std::make_unique<const SIGNAL::SignalDescriptor>(DummySignal::DUMMY_NAME, size, DummySignal::DUMMY_TYPE, DummySignal::DUMMY_HANDLE));
+
+    return {vDesc.back().get(),
```

Comment:
Same as above.

Reply:
Now, I fixed it for FSignal and EventSignal tests. In the case of FSignal and EventSignal the descriptors were constant so I used 'static const' and used one instance for all tests. For the integer signals the descriptor isn't constant. Initially the function was returning a test signal by value. Now, with memory optimizations the signal contains a pointer to its descriptor and it must be available somewhere. For integer signal tests the descriptors differ and 'static const' cannot be used as they differ by value. So the idea was to add a proper descriptor to the static container once the new test integer signal is created. It is only used once by this specific test. Finally when the test terminates all descriptors are destroyed.